### ISSUE TITLE
Not detect all new file in shared/group folders

### ISSUE THREAD

**Issue by mrmipo** [matiasdelellis/facerecognition#765]
Hi 

I use group folder for keeping family photo with my wife (auto upload from phone to selected group folder). But face recognize process only file that was upload by one user. To process all new file I need to execute :
     ` occ face:background_job --sync-mode `
what take loot of time (group folder have around 42k files). Same situation are with shared folder (I tested after notice that group folder has that issue)

### Steps to reproduce
1.  crate and enable group folder scanning
2.  enable face recognition for user A
3. upload photo to group folder by user A
4. check that if photo are process (`occ face:background_job -vvv -u A`)
5. upload photo to group folder by user B
6. check that if photo are process (`occ face:background_job -vvv -u A`)

### Server configuration

* **Operating system:** 
Nextcloud AIO latest

* **Pdlib version:**  
from matiasdelellis/facerecognition-external-model:v1

* **How is DLib installed:** Make sure it is working correctly with this [tool](https://github.com/matiasdelellis/pdlib-min-test-suite) 
matiasdelellis/facerecognition-external-model:v1

* **How is PDlib installed:** Make sure it is working correctly with this [tool](https://github.com/matiasdelellis/pdlib-min-test-suite) 
matiasdelellis/facerecognition-external-model:v1

* **PHP version:** 
PHP 8.2.20 (cli) (built: Jun 21 2024 01:33:27) (NTS)

* **Web server:**
Server version: Apache/2.4.59 (Unix)
Server built:   Jun 20 2024 20:55:11

* **Database:**
postgres (PostgreSQL) 16.3

* **Nextcloud version:**
Nextcloud Hub 8 (29.0.3)

* **facerecognition settings:**
`occ config:list facerecognition`

> {
>     "apps": {
>         "facerecognition": {
>             "analysis_image_area": "8167200",
>             "enabled": "yes",
>             "assigned_memory": "3221225472",
>             "installed_version": "0.9.51",
>             "types": "filesystem",
>             "model": "5",
>             "sensitivity": "0.4",
>             "handle_group_files": "true",
>             "min_confidence": "0.7",
>             "handle_shared_files": "true",
>             "handle_external_files": "true",
>             "clustering_batch_size": 5000,
>             "min_faces_in_cluster": "10"
>         }
>     }
> }

**Comment by vwbusguy** [comment]
Yeah, I have a cron job that keeps setting `full_image_scan_done false` on my user, since that's the only way I've found for it to keep clustering pictures in shared folders.
